src/MergeContext.js

```diff
@@ -954,12 +962,50 @@ class PullRequest {
         return false;
     }
 
+    async _getMergeCommit() {
+        const mergeSha = await GH.getReference("pull/" + this._prNumber() + "/merge");
+        return await GH.getCommit(mergeSha);
+    }
+
+    // obtains abandoned staging checks
+    async _getAbandonedStagingStatuses() {
+        assert(!this._stagedSha());
+
+        const allEvents = await GH.getIssueEvents(this._prNumber());
+        // staging events are PR events where the bot user created a commit referencing this PR
+        let stagingEvents = allEvents.filter(ev => ev.event === "referenced" && ev.actor.login === Config.githubUserLogin());
+        if (!stagingEvents.length)
+            return;
+
+        // just in case: events should be already in chronological order already
+        stagingEvents = stagingEvents.sort((ev1, ev2) => Date.parse(ev1.created_at) - Date.parse(ev2.created_at));
```

Comment:
>  GitHub guarantees chronological order

That is how it works, but I could not find this guarantee in the documentation.

> we may be reordering same-second events by sorting them

Do you think that sorting of equal items reorders them? I would not expect changes in case...

Reply:
> Do you think that sorting of equal items reorders them?

In many cases, it does. The question is whether this particular sort() is stable.

> BTW, it is [documented](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort) that the original order remains unchanged if elements are equal.

... starting with ES10. We are relying on ES6 support IIRC, but it is possible that the specific nodejs implementation we are using today is compliant with that specific ES10 requirement. Unfortunately, https://node.green/ does not detail/check this part AFAICT -- it only checks whether Array.sort() is available. An SO [comment](https://stackoverflow.com/questions/56063449/sorting-in-nodejs-not-stable-implications-for-memory-usage#comment98767764_56063449) says "[stable sort] was implemented first in Node 11. sort() is not stable in Node 10 or below".

What nodejs version are we using?

Can we assert somehow that it's sort() is stable or that it is compliant with ES10?

Reply:
Our nodejs version is 10.19. It [seems](https://github.com/nodejs/node/issues/29446) that sort stability starts from 11 version as you noted. We can update to 11 (and assert this requirement), but I could not find how to check sort() stability independently (or its compliancy with  ES10).

Reply:
I suggest asserting that the events are sorted instead of sorting them, with a comment explaining that we should not sort because sort() is unstable until ES10.

Reply:
Done (7bf03bc).